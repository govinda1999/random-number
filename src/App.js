import React,{Component} from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import Rule from './component/Rule'
import UserInput from './component/UserInput';
import NavBar from './component/NavBar'
import Result from './component/Result'
export default class App extends Component{
  constructor(props){
    super(props);
    this.state = {
      number:0,
      text:null,
      class:""
    }
  }

  componentDidMount  = () =>{
    this.setState({
      number:Math.floor(Math.random()*100+1) 
    },()=> console.log(this.state.number));
  }
  //to handle submit button
  handlesubmit = (value) =>{
    //error handling
    if(isNaN(value)||value===""){
      this.setState({text:null});
      window.alert("Enter Proper Number");
    }
    //check for condition
    if(Math.abs(this.state.number-parseInt(value))===0){
      this.setState({class:"text-success",text:"Success!!"});
    }
    else if(Math.abs(this.state.number-parseInt(value))<=4){
      this.setState({class:"text-danger",text:"Hot!!"});
    }
    else if(Math.abs(this.state.number-parseInt(value))<=15){
      this.setState({class:"text-warning",text:"Warm!!"});
    }
    else if(Math.abs(this.state.number-parseInt(value))>15){
      this.setState({class:"text-primary",text:"Cold!!"});
    }
  }

  render(){
    return(
      <div className="container" >
        <NavBar />
        <h1 className="text-center text-info mt-5 pt-5">Number Guessing Web App</h1>
        <div className="row justify-content-center mt-5">
          <div className="col-5">
            <UserInput handlesubmit={this.handlesubmit}/>
            <Result text={this.state.text} class={this.state.class}/>
          </div>
        </div>
        <Rule />
      </div>
    );
  }
}
