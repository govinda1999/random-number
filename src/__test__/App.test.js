import React from 'react';
import { unmountComponentAtNode,render} from "react-dom";
import App from '../App';
import { act } from "react-dom/test-utils";

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('App component render properly',()=>{
	act(() => {
		render(<App />, container);  
	});
	expect(container.textContent)
	.toBe("WebAppNumber Guessing Web AppSubmitRules:Enter number between 1-100.if Difference is zero => Success.if Difference is less than 4 => Hot.if Difference is less than 15 => Warm.if Difference is greater than 15 => Cold.");
})

