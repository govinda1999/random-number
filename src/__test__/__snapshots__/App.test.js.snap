// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component render properly 1`] = `"<div class=\\"container\\"><nav class=\\"navbar navbar-dark fixed-top bg-dark flex-md-nowrap p-0 shadow\\"><a class=\\"navbar-brand col-sm-3 col-md-2 mr-0\\" href=\\"http://localhost:3000\\">WebApp</a></nav><h1 class=\\"text-center text-info mt-5 pt-5\\">Number Guessing Web App</h1><div class=\\"row justify-content-center mt-5\\"><div class=\\"col-5\\"><input class=\\"w-100 border border-top-0 border-left-0 border-right-0 border-info rounded pl-2\\" type=\\"number\\" placeholder=\\"Enter number\\" min=\\"1\\" max=\\"100\\" name=\\"value\\" data-testid=\\"input_id\\" value=\\"\\"><button class=\\"w-100 btn btn-outline-primary mt-3\\" data-testid=\\"submit_id\\">Submit</button></div></div><p class=\\"h4 mt-3 pt-3\\">Rules:</p><ul class=\\"text-capitalize\\"><li>Enter number between 1-100.</li><li>if Difference is zero =&gt; Success.</li><li>if Difference is less than 4 =&gt; Hot.</li><li>if Difference is less than 15 =&gt; Warm.</li><li>if Difference is greater than 15 =&gt; Cold.</li></ul></div>"`;
